### PR TITLE
Load analyzer assemlbies in their own AssemblyLoadContext

### PR DESCRIPTION
**Pull request by JoeRobich**
Attempt to address https://github.com/dotnet/format/issues/734

### CODE REVIEW

File: src/Analyzers/AnalyzerReferenceInformationProvider.cs [hunk 1]
```diff
@@ -1,35 +1,94 @@
...
+                    if (File.Exists(possibleAssemblyPath))
+                    {
+                        return LoadFromAssemblyPath(possibleAssemblyPath);
+                    }
+                }
+                catch { }
+
+                // Try to load the requested assembly from the default load context.
+                return AssemblyLoadContext.Default.LoadFromAssemblyName(assemblyName);
+            }
```
Comment (JoeRobich) [inline src/Analyzers/AnalyzerReferenceInformationProvider.cs:91]: This nees to be better. =)

Comment (jmarolf) [reply src/Analyzers/AnalyzerReferenceInformationProvider.cs:91]: its a great start!

Comment (JoeRobich) [reply src/Analyzers/AnalyzerReferenceInformationProvider.cs]: Updated to use the same logic as the LooseAssemblyVersionLoader

**Review by jmarolf — COMMENTED** [review]
:)

**Pull request merged**